onnxruntime/core/providers/cpu/ml/tree_ensemble_common.h

```diff
@@ -0,0 +1,706 @@
+                       ? root->truenode
+                       : root->falsenode;
+          }
+        } else {
+          while (root->is_not_leave) {
+            val = x_data[root->feature_id];
+            root = val <= root->value ? root->truenode : root->falsenode;
+          }
+        }
+        break;
+      case NODE_MODE::BRANCH_LT:
+        TREE_FIND_VALUE(<)
+        break;
+      case NODE_MODE::BRANCH_GTE:
+        TREE_FIND_VALUE(>=)
+        break;
+      case NODE_MODE::BRANCH_GT:
+        TREE_FIND_VALUE(>)
+        break;
+      case NODE_MODE::BRANCH_EQ:
+        TREE_FIND_VALUE(==)
+        break;
+      case NODE_MODE::BRANCH_NEQ:
+        TREE_FIND_VALUE(!=)
+        break;
+      case NODE_MODE::LEAF:
+        break;
+    }
+  } else {  // Different rules to compare to node thresholds.
+    OTYPE threshold;
+    while (root->is_not_leave) {
+      val = x_data[root->feature_id];
+      threshold = root->value;
+      switch (root->mode) {
+        case NODE_MODE::BRANCH_LEQ:
+          root = val <= threshold || (root->is_missing_track_true && _isnan_(val))
+                     ? root->truenode
+                     : root->falsenode;
+          break;
+        case NODE_MODE::BRANCH_LT:
+          root = val < threshold || (root->is_missing_track_true && _isnan_(val))
+                     ? root->truenode
+                     : root->falsenode;
+          break;
+        case NODE_MODE::BRANCH_GTE:
+          root = val >= threshold || (root->is_missing_track_true && _isnan_(val))
+                     ? root->truenode
+                     : root->falsenode;
+          break;
+        case NODE_MODE::BRANCH_GT:
+          root = val > threshold || (root->is_missing_track_true && _isnan_(val))
+                     ? root->truenode
+                     : root->falsenode;
+          break;
+        case NODE_MODE::BRANCH_EQ:
+          root = val == threshold || (root->is_missing_track_true && _isnan_(val))
+                     ? root->truenode
+                     : root->falsenode;
+          break;
+        case NODE_MODE::BRANCH_NEQ:
+          root = val != threshold || (root->is_missing_track_true && _isnan_(val))
+                     ? root->truenode
+                     : root->falsenode;
+          break;
+        case NODE_MODE::LEAF:
+          break;
+      }
+    }
+  }
+  return root;
+}
+
+template <typename ITYPE, typename OTYPE>
+class TreeEnsembleCommonClassifier : TreeEnsembleCommon<ITYPE, OTYPE> {
+ protected:
+  bool weights_are_all_positive_;
+  bool binary_case_;
+  int64_t class_count_;
+  std::vector<std::string> classlabels_strings_;
+  std::vector<int64_t> classlabels_int64s_;
+  std::vector<int64_t> class_labels_;
+
+ public:
+  TreeEnsembleCommonClassifier(int omp_tree, int omp_N) : TreeEnsembleCommon<ITYPE, OTYPE>(omp_tree, omp_N) {}
+
+  inline int64_t get_class_count() const { return class_count_; }
+
+  void init(
+      const std::string& aggregate_function,
```

Comment:
Can this be done in the ctor instead of having a separate 'init' method?

Reply:
fixed